File: 2018/day5/part1.exs
File.read!("input.txt")
|> String.trim()
|> String.to_charlist()
|> Enum.reduce([], fn
  x, [] -> [x]
  x, [h | t] when abs(x - h) == 32 -> t
  x, acc -> [x | acc]
end)
|> length
|> IO.inspect()

File: 2018/day5/part1.rb
input = File.read("input.txt").chomp.chars

result = input.reduce("") do |acc, x|
  last = acc[-1] || ""
  if last.downcase == x.downcase && (last.ord - x.ord).abs == 32
    acc[0...-1]
  else
    acc + x
  end
end

puts result.length
